airbyte-integrations/connectors/source-amplitude/source_amplitude/streams.py

```diff
@@ -206,3 +173,26 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
 
     def path(self, **kwargs) -> str:
         return f"{self.api_version}/export"
+
+    def get_error_handler(self) -> ErrorHandler:
+        error_mapping = DEFAULT_ERROR_MAPPING | {
+            400: ErrorResolution(
+                response_action=ResponseAction.FAIL,
+                failure_type=FailureType.config_error,
+                error_message="The file size of the exported data is too large. Shorten the time ranges and try again. The limit size is 4GB.",
+            ),
+            403: ErrorResolution(
+                response_action=ResponseAction.FAIL,
+                failure_type=FailureType.config_error,
+                error_message="Access denied due to lack of permission or invalid API/Secret key or wrong data region.",
+            ),
+            404: ErrorResolution(
+                response_action=ResponseAction.IGNORE, failure_type=FailureType.config_error, error_message="No data collected."
+            ),
+            504: ErrorResolution(
+                response_action=ResponseAction.FAIL,
+                failure_type=FailureType.system_error,
```

Comment:
synced on this a bit:
- Based on the description this feels more like `FailureType.transient_error` and that we should be attempting retries.
- Same for the below manifest, we want this to be consistent w/ the error handler in the manifest as well

It does give me some pause that this might be more of a config error based on the previous message especially if this error is consistently reproduced. But we can always switch this after you look back on previous commits when this 504 catch block was introduced or what we observe after merging

Reply:
HTTP Status code/message mapping comes directly from Amplitude here: https://amplitude.com/docs/apis/analytics/export#status-codes

Therefore, I've updated the 504 mapping to FAIL as a config_error. I've added additional context for the user to reference Amplitude docs to set up the S3 destination.